internal/controller/utils.go

```diff
@@ -189,6 +203,36 @@ func parseHFURI(modelArtifact *msv1alpha1.ModelArtifacts) (string, string, error
 	return parts[0], parts[1], nil
 }
 
+// parseOCIURI returns parts from a valid oci URI, or
+// returns an error if the OCI URI is invalid
+// returns two strings
+// First string is the complete image identifier including tag
+// Second string is the path to the model
+func parseOCIURI(modelArtifact *msv1alpha1.ModelArtifacts) (string, []string, error) {
+	var imageIdentifier string
+	pathToModel := []string{}
+	if modelArtifact == nil {
+		return imageIdentifier, pathToModel, fmt.Errorf("modelArtifact is nil")
+	}
+
+	uri := modelArtifact.URI
+	if !isOCIURI(uri) {
+		return imageIdentifier, pathToModel, fmt.Errorf("URI does not have oci prefix: %s", uri)
+	}
+
+	// Split by ::
+	parts := strings.Split(strings.TrimPrefix(uri, MODEL_ARTIFACT_URI_OCI_PREFIX), ociPathToModelSep)
+	if len(parts) != 2 {
+		return imageIdentifier, pathToModel, fmt.Errorf("invalid oci URI format: %s; need oci://<image identifier with tag>::/path/to/model. Please double check if you are missing %s which did not resul in exactly two segments", uri, ociPathToModelSep)
```

Comment:
s/resul/result

Reply:
Hi @rphillips not sure if this comment is a mistake?

Reply:
There is a typo in the Errorf() message.

Reply:
Got it thanks for catching it! I didn't understand the what `s/result/result` meant.